### PR TITLE
Training prep

### PR DESCRIPTION
**Pull request by jburel**
In this PR
* Update the README to match the new workflow
* Update script to handle new parameter git_user
* Introduce new steps
* Remove infrastructure dependency
* Fix build failure (``setuptools``)
* Use Groovy Sandbox selected see https://wiki.jenkins.io/display/JENKINS/Script+Security+Plugin
* No longer required to have the ome/script repo forked i.e. use ``--shallow`` for the merge command of OMERO-push

To test this PR:

* Follow the instructions in the README and deploy in OpenStack. To read the readme
and troubleshooting file it is easier to read them from https://github.com/jburel/devspace/tree/readme-training where they are rendered by gh

@pwalczysko if you want to try again, before you start you will have to
 * organize the ssh keys and gitconfig files in a specific folder (no need to recreate key and token)
 * run in devspace directory rm -rf ansible/vendor/openmicroscopy.devspace to be sure that you get the latest version of the role
The playbook to provision the devspace now only needs to be run once. I fixed the issue

@sbesson since we had to have a "specific" gitconfig file, I went a step further and remove the paramater ``github_user`` and require ``user`` in the gitconfig file. The value will be read from the file.
This means that user will only have to specify the path to the repo containing the ssh key and config files.

--depends-on https://github.com/openmicroscopy/ansible-role-devspace/pull/5

### CODE REVIEW

**Comment by pwalczysko** [comment]
``The following steps are only need to be done the first time you want to generate instances.`` 

from https://github.com/jburel/devspace/tree/readme-training should be

 ``The following steps are only needed to be done the first time you want to generate instances.`` 

The sentence occurs twice in that README (``needed`` instead of ``need``)

**Comment by pwalczysko** [comment]
```
Register a key: Go to Access & Security > Key Pairs and click on Import Key Pair. Copy the content of the public key you use to access our resources e.g. id_rsa.pub
```
After clicking on Access & Security, you have to first click onto ``Key Pairs`` tab. Then you can click on ``Import Key Pair`` button

**Comment by pwalczysko** [comment]
```


Create a file devspace_config/.ssh/config and add the following:

  Host github.com
      User git
      IdentityFile ~/.ssh/id_gh_rsa

```

This looks like a mistake to me. I do not have any IdentityFile under ``~/.ssh/id_gh_rsa``. Instead, I have this file under ``path/to/devspace_config/.ssh/id_gh_rsa``

**Comment by pwalczysko** [comment]
```
[github]
          token = your_token
          user = your_gh_username
  [user]
          email = your_email_address
          name = your_name
```
What is ``your_name`` here ? Is it my normal citizen name (first_name and surname ?) ?

**Comment by joshmoore** [comment]
Real name, yes. It shows up in the commit messages as: "your_name <your_email_address>" a la:

```
$ git log -n 1 | grep Author
Author: jmoore <josh@glencoesoftware.com>
```

**Comment by pwalczysko** [comment]
```


    Create an instance on OpenStack using the playbook create-devspace.yml. It is recommended to prefix the name of the devspace by your name or your initals:

      (dev) $ ansible-playbook create-devspace.yml -e vm_name=your_name-devspace-name -e vm_key_name=your_key

```

What is ``your_key`` here please ?

**Comment by joshmoore** [comment]
> What is your_key here please ?

The name of the key you created on or uploaded to openstack.

**Comment by jburel** [comment]
https://github.com/openmicroscopy/devspace/pull/81#issuecomment-344228965.
This is correct it will be used on devspace not locally

**Comment by pwalczysko** [comment]
In the ``Access the devspace`` paragraph of the README https://github.com/jburel/devspace/tree/readme-training, I would have two suggestions:

  - is the url ``https://devspace_openstack_ip:$PORT`` actually a Command ? (it is highlighted as Command in the table in the abovementioned paragraph -> this is confusing
  - after I ran the ``docker-compose port $SERVICE $PRIVATE_PORT``, I got a slightly non-trivial output like 

```
docker-compose port nginx 80
WARNING: The JENKINS_PASSWORD variable is not set. Defaulting to a blank string.
WARNING: The USER_ID variable is not set. Defaulting to a blank string.
0.0.0.0:xxxxxx
```
where xxxxx is my port and I should substitute it for the $PORT variable which is in the url ``https://devspace_openstack_ip:$PORT`` -> maybe please explain this step, it would spare me trying ``https://devspace_openstack_ip:80``

**Comment by pwalczysko** [comment]
But my build is failing, see https://10.0.51.143:32769/job/BIOFORMATS-push/2/console.

**Comment by jburel** [comment]
Do you have a fork of Bio-Formats?

**Comment by jburel** [comment]
Failure was due to a misconfiguration in the config file
@pwalczysko has now his devspace up and running.
I will clarify in the readme that the files are not used locally to avoid confusion.

**Comment by jburel** [comment]
@pwalczysko I have updated the readme and troubleshooting and clarify the points we discussed
When you have a change, could you read through?
no need to redeploy a devspace

File: ansible/requirements.yml [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+- src: openmicroscopy.mysql-backup
+  version: 0.1.0
+
+- src: openmicroscopy.omero-web-apps
+  version: 0.1.0
+
+- src: openmicroscopy.reboot-server
+  version: 0.1.0
+
+- src: git+https://github.com/jburel/ansible-role-devspace.git
```
Comment (sbesson) [inline ansible/requirements.yml]: @jburel now everything has been tested by @pwalczysko  and https://github.com/openmicroscopy/ansible-role-devspace/pull/5 has been integrated, can you push another commit to point at the master branch of the upstream repo for now

**Comment by joshmoore** [comment]
Bumped, @sbesson

**Pull request merged**